src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs

```diff
@@ -251,6 +253,65 @@ public IActionResult Recover([FromBody]WalletRecoveryRequest request)
                 this.logger.LogError("Exception occurred: {0}", e.ToString());
                 return ErrorHelpers.BuildErrorResponse(HttpStatusCode.BadRequest, e.Message, e.ToString());
             }
+            finally
+            {
+                this.logger.LogTrace("(-)");
+            }
+        }
+
+        /// <summary>
+        /// Recovers a wallet using only the extended public key.
+        /// </summary>
+        /// <param name="request">The object containing the parameters used to recover a wallet.</param>
+        [Route("recover-via-extpubkey")]
+        [HttpPost]
+        public IActionResult RecoverViaExtPubKey([FromBody]WalletExtPubRecoveryRequest request)
+        {
+            Guard.NotNull(request, nameof(request));
+            this.logger.LogTrace("({0}.{1}:'{2}')", nameof(request), nameof(request.Name), request.Name);
+
+            if (!this.ModelState.IsValid)
+            {
+                this.logger.LogTrace("(-)");
```

Comment:
Inside the same method the exit logs should be different to be able to distinguish execution path based on the exit log

Reply:
ok